content/developer/release-and-packaging/use-a-versionining-convention.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: practice
+weight: 1
+title: Use a versioning convention
+what: >
+  A system in place to label your versions.
+why: >
+  This helps to determine how your software is released and provides clarity to users on recency
+when: Mid-Development
+importance: Medium
+implementations:
+  - desc: SemVer
+    advantages:
+      - Provides a standard way of communicating whether changes were major, minor, or patches
+      - Provides compatibility guarantees within major versions
+    disadvantages:
+      - What is determined to be major, minor, or patch items can be arbitrary between projects and can be difficult to define
+      - Can be complex when working with larger projects
+  - desc: CalVer
+    advantages:
+      - Clear time-based context
+      - Codifies release cadence
+    disadvantages:
+      - Less detail on changes
+      - No built-in compatibility guarantees
```

Comment:
Another challenge here is around being clear what you versions have long-term support. I remember a discussion about this when Dask switched to CalVer; it was hard to determine what would be considered the "blessed" version to support. I think larger projects do this by very strictly sticking to a (slower) release cadence for major changes; this can effectively make it harder to move quickly.

Not sure exactly how to phrase that succinctly, and maybe it is beyond scope for this, but something to consider.